### PR TITLE
Add NATS example 

### PR DESCRIPTION
**Pull request by aashikam**
## Purpose

Fixes: https://github.com/ballerina-platform/ballerina-standard-library/issues/2215

## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [x] Added tests

### CODE REVIEW

File: examples/order-manager/notification_service/notification_service.bal [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+// under the License.
+
+import ballerinax/nats;
+import ballerina/lang.value;
+import ballerina/log;
+import notification_service.types;
+
+configurable string LISTENING_SUBJECT = ?;
+
+listener nats:Listener subscription = new(nats:DEFAULT_URL);
```
Comment (shafreenAnfar) [inline examples/order-manager/notification_service/notification_service.bal:24]: We can have it inline right ?

File: examples/order-manager/notification_service/notification_service.bal [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+@nats:ServiceConfig {
+    subject: LISTENING_SUBJECT
+}
+service nats:Service on subscription {
```
Comment (shafreenAnfar) [inline examples/order-manager/notification_service/notification_service.bal:29]: `nats:Service` is optional and can be removed right ?

File: examples/order-manager/notification_service/notification_service.bal [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    // Listens to NATS subject for any successful orders
+    remote function onMessage(nats:Message message) returns error? {
+
+        // Convert the byte values in the NATS Message to type Order
+        string messageContent = check string:fromBytes(message.content);
+        json content = check value:fromJsonString(messageContent);
+        json jsonTweet = content.cloneReadOnly();
```
Comment (shafreenAnfar) [inline examples/order-manager/notification_service/notification_service.bal:37]: Is the variable name `jsonTweet` correct ?

File: examples/order-manager/order_processor/order_processor.bal [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+import ballerinax/nats;
+import order_processor.types;
+import ballerina/lang.value;
+import ballerina/log;
+
+configurable string LISTENING_SUBJECT = ?;
+configurable string PUBLISH_SUBJECT = ?;
+
+// Creates a NATS client with default configurations
+nats:Client natsClient = check new(nats:DEFAULT_URL);
```
Comment (shafreenAnfar) [inline examples/order-manager/order_processor/order_processor.bal:26]: Does not have to be a module level varaible.

File: examples/order-manager/order_processor/order_processor.bal [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+
+listener nats:Listener subscription = new(nats:DEFAULT_URL);
```
Comment (shafreenAnfar) [inline examples/order-manager/order_processor/order_processor.bal:28]: Can have it inline.

File: examples/order-manager/order_processor/order_processor.bal [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+@nats:ServiceConfig {
+    subject: LISTENING_SUBJECT
+}
+service nats:Service on subscription {
+
+    // Listens to NATS subject for any new orders and process them
+    remote function onMessage(nats:Message message) returns error? {
+
+        // Uses Ballerina query expressions to filter out the successful orders and publish to Kafka topic
+        error? err = from types:Order 'order in check getOrdersFromRecords(message) where 'order.status == types:SUCCESS do {
```
Comment (shafreenAnfar) [inline examples/order-manager/order_processor/order_processor.bal:39]:
```ballerina
check from types:Order 'order in check getOrdersFromRecords(message) 
where 'order.status == types:SUCCESS 
do {
     log:printInfo("Sending successful order to " + PUBLISH_SUBJECT + " " + 'order.toString());
     // Publish the order to the NATS subject
     check natsClient->publishMessage({
          content: 'order.toString().toBytes(),
          subject: PUBLISH_SUBJECT
     });
 };
```

Comment (shafreenAnfar) [reply examples/order-manager/order_processor/order_processor.bal:39]: Lets format it as above.

File: examples/order-manager/order_service/order_service.bal [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+        check publishOrder(randomOrder);
+        return "Message sent to the NATS subject " + SUBJECT + " successfully. Order " + message + " with status " + paymentStatus;
+    }
+}
+
+function publishOrder(types:Order 'order) returns error? {
+    log:printInfo("Publishing order " + 'order.toString());
+
+    // Publish the order to the NATS subject
+    check natsClient->publishMessage({
```
Comment (shafreenAnfar) [inline examples/order-manager/order_service/order_service.bal:59]:
```ballerina
check natsClient->publishMessage(
  {
    content: 'order.toString().toBytes(),
    subject: SUBJECT
  }
);
```

File: examples/order-manager/order_service/tests/order_service_tests.bal [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+    }
+}
+
+isolated function getMessageReceived() returns string {
+    lock {
+        return messageReceived;
+    }
+}
+
+configurable string username = "user";
```
Comment (shafreenAnfar) [inline examples/order-manager/order_service/tests/order_service_tests.bal:39]: We have used uppercase in other files. Shall we make it consistent.

File: examples/order-manager/order_processor/order_processor.bal (hunk 4 above)
Comment (aashikam) [reply examples/order-manager/order_processor/order_processor.bal:26]: If I take this inside `onMessage`, for every message a client will be created.

Comment (shafreenAnfar) [reply examples/order-manager/order_processor/order_processor.bal:26]: What I meant was to make it an object level variable.

File: examples/order-manager/notification_service/notification_service.bal [hunk 9]
```diff
@@ -21,21 +21,19 @@ import notification_service.types;
 
 configurable string LISTENING_SUBJECT = ?;
 
-listener nats:Listener subscription = new(nats:DEFAULT_URL);
-
 @nats:ServiceConfig {
     subject: LISTENING_SUBJECT
 }
-service nats:Service on subscription {
+service nats:Service on new nats:Listener(nats:DEFAULT_URL) {
```
Comment (shafreenAnfar) [inline examples/order-manager/notification_service/notification_service.bal:27]: service "notificationService" on new nats:Listener(nats:DEFAULT_URL) 

Can't we do the above ?

**Review by shafreenAnfar — APPROVED** [review]

**Pull request merged**